ledger_analytics/forecast.py

```diff
@@ -0,0 +1,252 @@
+        """AR1 model configuration class.
+
+        Attributes:
+            loss_family: the likelihood family to use. One of ``"Gamma"``, ``"Lognormal"``,
+                ``"Normal"`` or ``"InverseGaussian"``. Defaults to ``"Gamma"``.
+            loss_definition: the field to model in the triangle. One of
+                ``"paid"`` ``"reported"`` or ``"incurred"``.
+            recency_decay: geometric decay parameter to downweight earlier
+                diagonals (see `Modeling rationale...` section
+                in the User Guide). Defaults to 1.0 for no geometric decay.
+                Can be ``"lookup"`` to choose based on ``line_of_business``.
+            priors: dictionary of priors. Defaults to ``None`` to use the default priors.
+                See the DefaultPriors class for default (non line-of-business)
+                priors.
+            autofit_override: override the MCMC autofitting procedure arguments. See the documentation
+                for a fully description of options in the User Guide.
+            prior_only: should a prior predictive simulation be run?
+            seed: Seed to use for model sampling. Defaults to ``None``, but it is highly recommended
+                to set.
+        """
+
+        loss_family: Literal["Gamma", "Lognormal", "Normal", "InverseGaussian"] = (
+            "Gamma"
+        )
+        loss_definition: Literal["paid", "reported", "incurred"] = "paid"
+        recency_decay: str | float | None = None
+        priors: dict[str, list[float] | float] | None = None
+        autofit_override: dict[str, float | int | None] = None
+        prior_only: bool = False
+        seed: int | None = None
+
+    class PredictConfig(ValidationConfig):
+        """AR1 predict configuration class.
+
+        Attributes:
+            include_process_risk: should process risk or
+                aleatoric uncertainty be included in the predictions.
+                Defaults to ``True``. If ``False``, predictions are
+                based on the mean function, only.
+        """
+
+        include_process_risk: bool = True
+
+
+class SSM(ForecastModel):
+    """SSM (State Space Model).
+
+    This model is a, by default, Bayesian state space model or
+    Bayesian structural time series model similar to a local linear trend
+    model with mean reversion (with drift) and a 'momentum' term which
+    is analogous to a moving average lag-1 component with decay.
+    The model has the form:
+
+    ..  math::
+
+        \mathrm{LR}_{i} &\sim \mathrm{Gamma}(\exp(\eta_{i}), \\sigma_{i}^2)\\\\
+        \eta_{i} &= (1 - \phi_{\mathrm{reversion}}) \mathrm{LR}_{\mathrm{target}} + \phi_{\mathrm{reversion}} \eta_{i - 1} + \zeta_{i-1} + z_{i} \sqrt{\epsilon_{\mathrm{latent}}}\\\\
+        \zeta_{i} &= \gamma_{\mathrm{momentum}} (\zeta_{i-1} + z_{i} \sqrt{\epsilon_{\mathrm{latent}}})\\\\
+        \\sigma_{i}^2 &= \exp(\\sigma_{\mathrm{base}})^2 + \exp(\\sigma_{\mathrm{obs}})^2 / \sqrt{\mathrm{EP}_i}
```

Comment:
Should we allude to used earned premium here since cape cod is the default? 
```suggestion
        \\sigma_{i}^2 &= \exp(\\sigma_{\mathrm{base}})^2 + \exp(\\sigma_{\mathrm{obs}})^2 / \sqrt{\mathrm{UEP}_i}
```